### DISCUSSION TITLE
cannot change array elements through slicing

### DISCUSSION THREAD

**Discussion by SleepyRoy** [vlang/v#11690] [Ideas]
**V version:** 
latest master with mingw64, msvc, and tcc

**OS:** 
win10

**What did you do?**
```V
fn main() {
    mut arr := [1,2,3,4]
    arr[1..3] = [0,0]
    println(arr)
} 
```

**What did you expect to see?**
[1,0,0,4]
 
**What did you see instead?**
Interestingly, results are different with different backends.

TCC:
compiled but printed [1,2,3,4]

mingw64, msvc:
C error

**Comment by yuyi98** [comment]
Should such assignments be considered a syntax error?

**Comment by SleepyRoy** [comment]
@yuyi98 Actually I'm not very sure. If a slice is not a copy, then I guess such syntax should be allowed.

**Comment by JalonSolov** [comment]
Pretty certain slices are copies right now.  There's an issue open right now discussing adding 'views', so that code like the example here would work.

**Comment by ntrel** [comment]
@JalonSolov Do you have a link to the discussion? This example suggests slicing does not make a copy:
```v
fn main() {
    mut arr := [1,2,3,4]
    mut s := arr[1..3]
    s[0] = 6
    assert arr[1] == 6 // passes
} 
```

**Comment by JalonSolov** [comment]
#5480

**Comment by danieldaeschle** [comment]
```
main.v:3:5: error: cannot reassign using range expression on the left side of an assignment
    1 | fn main() {
    2 |     mut arr := [1, 2, 3, 4]
    3 |     arr[1..3] = [0, 0]
      |        ~~~~~~
    4 |     println(arr)
    5 | }
```

current behavior
